public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.ts

```diff
@@ -0,0 +1,610 @@
+    pluginId: panel.spec.vizConfig.kind,
+    options: panel.spec.vizConfig.spec.options,
+    fieldConfig: transformValueMappingsToV1(panel.spec.vizConfig.spec.fieldConfig),
+    pluginVersion: panel.spec.vizConfig.spec.pluginVersion,
+    // FIXME: Transparent is not added to the schema yet
+    // displayMode: panel.spec.transparent ? 'transparent' : undefined,
+    hoverHeader: !panel.spec.title && !timeOverrideShown,
+    hoverHeaderOffset: 0,
+    $data: createPanelDataProvider(panel),
+    titleItems,
+    $behaviors: [],
+    extendPanelContext: setDashboardPanelContext,
+    // _UNSAFE_customMigrationHandler: getAngularPanelMigrationHandler(panel), //FIXME: Angular Migration
+  };
+
+  // FIXME: Library Panel
+  // if (panel.spec.libraryPanel) {
+  //   vizPanelState.$behaviors!.push(
+  //     new LibraryPanelBehavior({ uid: panel.spec.libraryPanel.uid, name: panel.spec.libraryPanel.name })
+  //   );
+  //   vizPanelState.pluginId = LibraryPanelBehavior.LOADING_VIZ_PANEL_PLUGIN_ID;
+  //   vizPanelState.$data = undefined;
+  // }
+
+  if (!config.publicDashboardAccessToken) {
+    vizPanelState.menu = new VizPanelMenu({
+      $behaviors: [panelMenuBehavior],
+    });
+  }
+
+  if (queryOptions.timeFrom || queryOptions.timeShift) {
+    vizPanelState.$timeRange = new PanelTimeRange({
+      timeFrom: queryOptions.timeFrom,
+      timeShift: queryOptions.timeShift,
+      hideTimeOverride: queryOptions.hideTimeOverride,
+    });
+  }
+
+  return new VizPanel(vizPanelState);
+}
+
+function trackIfEmpty(grid: SceneGridLayout) {
+  getDashboardSceneFor(grid).setState({ isEmpty: grid.state.children.length === 0 });
+
+  const sub = grid.subscribeToState((n, p) => {
+    if (n.children.length !== p.children.length || n.children !== p.children) {
+      getDashboardSceneFor(grid).setState({ isEmpty: n.children.length === 0 });
+    }
+  });
+
+  return () => {
+    sub.unsubscribe();
+  };
+}
+
+function registerPanelInteractionsReporter(scene: DashboardScene) {
+  scene.subscribeToEvent(UserActionEvent, (e) => {
+    const { interaction } = e.payload;
+    switch (interaction) {
+      case 'panel-status-message-clicked':
+        DashboardInteractions.panelStatusMessageClicked();
+        break;
+      case 'panel-cancel-query-clicked':
+        DashboardInteractions.panelCancelQueryClicked();
+        break;
+    }
+  });
+}
+
+export function createPanelDataProvider(panelKind: PanelKind): SceneDataProvider | undefined {
+  const panel = panelKind.spec;
+  const targets = panel.data?.spec.queries ?? [];
+  // Skip setting query runner for panels without queries
+  if (!targets?.length) {
+    return undefined;
+  }
+
+  // Skip setting query runner for panel plugins with skipDataQuery
+  if (config.panels[panel.vizConfig.kind]?.skipDataQuery) {
+    return undefined;
+  }
+
+  let dataProvider: SceneDataProvider | undefined = undefined;
+  const datasource = panel.data.spec.queries[0]?.spec.datasource;
```

Comment:
In the new schema, there's no indication that a query is mixed (in the v1 panel has `datasource` prop set to mixed datasource). For this to work I think we need to figure out here if the queries in the panel spec have different datasources applied and set the datasource to mixed accordingly.